src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -0,0 +1,602 @@
+ */
+package org.ow2.proactive.connector.iaas.cloud.provider.azure;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.apache.log4j.Logger;
+import org.ow2.proactive.connector.iaas.cloud.provider.CloudProvider;
+import org.ow2.proactive.connector.iaas.model.Hardware;
+import org.ow2.proactive.connector.iaas.model.Image;
+import org.ow2.proactive.connector.iaas.model.Infrastructure;
+import org.ow2.proactive.connector.iaas.model.Instance;
+import org.ow2.proactive.connector.iaas.model.InstanceCredentials;
+import org.ow2.proactive.connector.iaas.model.InstanceScript;
+import org.ow2.proactive.connector.iaas.model.Options;
+import org.ow2.proactive.connector.iaas.model.ScriptResult;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.Lists;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.compute.OperatingSystemTypes;
+import com.microsoft.azure.management.compute.VirtualMachine;
+import com.microsoft.azure.management.compute.VirtualMachineCustomImage;
+import com.microsoft.azure.management.compute.VirtualMachineDataDisk;
+import com.microsoft.azure.management.compute.VirtualMachineExtension;
+import com.microsoft.azure.management.compute.VirtualMachineSizeTypes;
+import com.microsoft.azure.management.network.Network;
+import com.microsoft.azure.management.network.NetworkInterface;
+import com.microsoft.azure.management.network.NetworkSecurityGroup;
+import com.microsoft.azure.management.network.PublicIpAddress;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.utils.SdkContext;
+
+import lombok.Getter;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 01/03/17
+ */
+@Component
+public class AzureProvider implements CloudProvider {
+
+    private final Logger logger = Logger.getLogger(AzureProvider.class);
+
+    @Getter
+    private final String type = "azure";
+
+    private static final Region defaultRegion = Region.US_EAST;
+
+    private static final VirtualMachineSizeTypes defaultVMSize = VirtualMachineSizeTypes.STANDARD_D1_V2;
+
+    private static final int resourcesNameExtraChars = 5;
+
+    private static final String virtualNetworkNameBase = "vnet";
+
+    private static final String publicIPAddressNameBase = "ip";
+
+    private static final String networkSecurityGroupNameBase = "sg";
+
+    private static final String networkInterfaceNameBase = "if";
+
+    private static final String osDiskNameBase = "os";
+
+    private static final String defaultUsername = "activeeon";
+
+    private static final String defaultPassword = "Act1v€0N";
```

Comment:
The default username/password is just a way to bypass the need to provide credentials during instances creation.
It is not associated to existing account but used to setup/modify the credentails of the root/administrator user from the image during the deployment.
I chose this password because it must comply with some rules (minimal size, at least 1 special char & capital, ..) not to be rejected.